src/box/txn_limbo.c

```diff
@@ -1306,6 +1337,134 @@ txn_limbo_update_system_spaces_is_sync_state(struct txn_limbo *limbo,
 	system_spaces_update_is_sync_state(is_promote);
 }
 
+static int
+txn_limbo_entry_iterator_next(struct port_c_iterator *it, struct port *out,
+			      bool *is_eof)
+{
+	struct txn_limbo_entry_port_c_iterator *entry_it =
+		(struct txn_limbo_entry_port_c_iterator *)it;
+	struct txn_stmt *stmt = entry_it->stmt;
+	if (stmt == NULL) {
+		*is_eof = true;
+		return 0;
+	}
+
+	*is_eof = false;
+	struct xrow_header *row = stmt->row;
+	port_c_create(out);
+	struct region *region = &fiber()->gc;
+	size_t region_svp = region_used(region);
+	struct mpstream stream;
+	mpstream_init(&stream, region, region_reserve_cb, region_alloc_cb,
+		      mpstream_panic_cb, NULL);
+	mpstream_encode_map(&stream, 2);
+	mpstream_encode_str(&stream, "meta");
+	mpstream_encode_map(&stream, 3);
+	mpstream_encode_str(&stream, "replica_id");
+	mpstream_encode_uint(&stream, row->replica_id);
+	mpstream_encode_str(&stream, "lsn");
+	mpstream_encode_uint(&stream, row->lsn);
+	mpstream_encode_str(&stream, "term");
+	mpstream_encode_uint(&stream, txn_limbo.promote_greatest_term);
+	mpstream_encode_str(&stream, "statement");
+	struct request request;
+	VERIFY(xrow_decode_dml(row, &request,
+			       dml_request_key_map(row->type)) == 0);
+	size_t map_size = 2 + (request.key != NULL) +
+				     (request.tuple != NULL) +
+				     (request.ops != NULL) +
+				     (stmt->old_tuple != NULL) +
+				     (stmt->new_tuple != NULL);
+	mpstream_encode_map(&stream, map_size);
+	mpstream_encode_str(&stream, "op_type");
+	mpstream_encode_uint(&stream, row->type);
+	mpstream_encode_str(&stream, "space_id");
+	mpstream_encode_uint(&stream, space_id(stmt->space));
+#define encode_request_field(field_name)				    \
+do {									    \
+	if (request.field_name != NULL) {				    \
+		mpstream_encode_str(&stream, #field_name);		    \
+		size_t len = request.field_name##_end - request.field_name; \
+		char *buf = mpstream_reserve(&stream, len);		    \
+		memcpy(buf, request.field_name, len);			    \
+		mpstream_advance(&stream, len);				    \
+	}								    \
+}									    \
+while (0)
+	encode_request_field(key);
+	encode_request_field(tuple);
+	encode_request_field(ops);
+#undef encode_request_part
```

Comment:
`encode_request_part` is not defined anywhere. Did you mean `encode_request_field`?